backend/src/test/groovy/pt/ulisboa/tecnico/socialsoftware/tutor/user/service/CreateExternalUserTest.groovy

```diff
@@ -0,0 +1,171 @@
+        result.getState() == User.State.INACTIVE
+        and:"checks if the user and the course execution are associated"
+        result.getCourseExecutions().size() == 1
+        result.getCourseExecutions().get(0).getAcronym() == COURSE_1_ACRONYM
+        result.getCourseExecutions().get(0).getAcademicTerm() == COURSE_1_ACADEMIC_TERM
+        result.getConfirmationToken() != ""
+        courseExecution.getUsers().size() == 1
+        courseExecution.getUsers().toList().get(0).getId() == result.getId()
+        and: "a mail is sent"
+        1 * mailerMock.sendSimpleMail('pedro.test99@gmail.com', EMAIL,_,_)
+    }
+
+    def "the course execution exists, the username exists and associate the user with the course execution" (){
+        given: "a user"
+        def user2 = new User("", EMAIL, User.Role.STUDENT)
+        user2.addCourse(courseExecution)
+        courseExecution.addUser(user2)
+        userRepository.save(user2)
+        and: "a external course execution"
+        def executionId = courseExecution.getId()
+        and: "a external user dto"
+        externalUserDto = new ExternalUserDto()
+        externalUserDto.setEmail(EMAIL)
+        externalUserDto.setRole(User.Role.STUDENT)
+
+        when:
+        def result = userService.createExternalUser(executionId, externalUserDto)
+
+        then:"the user is saved in the database"
+        System.out.println(userRepository.findAll())
+        userRepository.count() == 4
+        and: "checks if user data is correct"
+        result.getUsername() == EMAIL
+        result.getEmail() == EMAIL
+        and:"checks if the user and the course execution are associated"
+        result.getCourseExecutions().size() == 1
+        result.getCourseExecutions().get(0).getAcronym() == COURSE_1_ACRONYM
+        result.getCourseExecutions().get(0).getAcademicTerm() == COURSE_1_ACADEMIC_TERM
+        result.getConfirmationToken() != ""
+        courseExecution.getUsers().size() == 1
+        courseExecution.getUsers().toList().get(0).getId() == result.getId()
+        and: "a mail is sent"
+        1 * mailerMock.sendSimpleMail('pedro.test99@gmail.com', EMAIL,_,_)
+    }
+
+    @Unroll
+    def "invalid arguments: email=#email | password=#password | role=#role"(){
+        given: "a invalid course execution id"
+        def executionId = courseExecution.getId()
+        and: "a external user dto"
+        externalUserDto = new ExternalUserDto()
+        externalUserDto.setEmail(email)
+
+        when:
+        userService.createExternalUser(executionId, externalUserDto)
+
+        then:
+        def error = thrown(TutorException)
+        error.getErrorMessage() == errorMessage
+        and: "no mail is sent"
+        0 * mailerMock.sendSimpleMail(_,_,_,_)
+
+        where:
+        email       | role                     || errorMessage
+        null        | User.Role.STUDENT        || ErrorMessage.INVALID_EMAIL
+        ""          | User.Role.STUDENT        || ErrorMessage.INVALID_EMAIL
```

Comment:
test with other roles, TEACHER and ADMIN....it should not be possible to create an admin external user